File: frontend/types/publicRealtor.ts
export type PublicRealtorProfile = {
  user_id: number
  full_name?: string | null
  agency_name?: string | null
  avatar_url?: string | null
  is_verified: boolean
  member_since?: string | null
  active_listings_count: number
  phone?: string | null
  whatsapp?: string | null
  telegram_username?: string | null
}

export type PublicRealtorProfilePage = {
  realtor: PublicRealtorProfile
  properties: {
    items: Array<{
      id: number
      title: string
      price: number | null
      city: string | null
      rooms: number | null
      image_url: string | null
      status: string
      last_verified_at?: string | null
    }>
    total: number
    limit: number
    offset: number
  }
}

File: backend/app/schemas/public_realtor.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.property import PropertyCardResponse, PropertyListResponse


class PublicRealtorProfileResponse(BaseModel):
    user_id: int
    full_name: str | None = None
    agency_name: str | None = None
    avatar_url: str | None = None
    is_verified: bool = False
    member_since: datetime | None = None
    active_listings_count: int = 0
    phone: str | None = None
    whatsapp: str | None = None
    telegram_username: str | None = None

    model_config = ConfigDict(from_attributes=True)


class PublicRealtorProfilePageResponse(BaseModel):
    realtor: PublicRealtorProfileResponse
    properties: PropertyListResponse
